frontend/src/views/Join.tsx

```diff
@@ -1,27 +1,121 @@
-import React from 'react';
+import React, { ReactElement, useEffect, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import EnterNicknamePage from '../components/core/EnterNickname';
+import { LargeCenterInputText, LargeInputButton } from '../components/core/Input';
+import { LargeText, Text } from '../components/core/Text';
 
 function JoinGamePage() {
   // Get history object to be able to move between different pages
   const history = useHistory();
 
   /**
-   * Redirect the user to the lobby.
+   * Page state variable that defines whether the user is
+   * entering a room ID or a nickname.
+   * 1 = Enter Room ID (default)
+   * 2 = Enter Nickname
+   */
+  const [pageState, setPageState] = useState(parseInt(localStorage.getItem('pageState') || '1', 10));
+
+  // Variable to hold the user's current room ID input (default empty string).
+  const [roomId, setRoomId] = useState(localStorage.getItem('roomId') || '');
+
+  /**
+   * The nickname is valid if it is non-empty and has exactly
+   * six nonalphanumeric characters.
+   */
+  const isValidRoomId = (roomIdParam: string) => (roomIdParam.length === 6) && /^\d+$/.test(roomIdParam);
+  const [validRoomId, setValidRoomId] = useState(false);
+  useEffect(() => {
+    setValidRoomId(isValidRoomId(roomId));
+  }, [roomId]);
+
+  // Variable to hold whether the user is focused on the text input field.
+  const [focusInput, setFocusInput] = useState(false);
+
+  /**
+   * The local storage is updated when the page state updates.
+   */
+  const updateLocalStorage = (pageStateParam: string, roomIdParam: string) => {
+    localStorage.setItem('pageState', pageStateParam);
+    localStorage.setItem('roomId', roomIdParam);
+  };
```

Comment:
This all looks great, although I'm also curious what your opinion would be if we decided not to save page state across refreshes? A couple of concerns I have now are 1. you can't leave the nickname state if you decide you want to switch rooms and 2. the page state persists in the browser, which might be unexpected if users wanted to join another room in a different tab or revisited the website after several days.

Not sure how Kahoot does things, but I'd honestly be fine with a page refresh simply resetting to state 1 (if users accidently refreshed they'd just need to re-enter the room id), or maybe when users enter a room id, it "redirects" them to `game/join/{id}` (using someting like [useParams](https://reactrouter.com/web/api/Hooks)) or `game/join?roomId=XXXXXX` (this is also a card in the sprint) which then shows the nickname display? That way the back button still works and it doesn't persist longer than a single session. 

My opinion might be the minority though, so I'd love to hear your thoughts as well as what @helenli8181 thinks, either here or possibly in a call as well.

Reply:
I think it would be fine to not save page state when refreshing. The number of possible errors we could avoid would be worth having to re-enter room code if you accidentally refresh imo.

Reply:
Yeah for sure, I have no problem with that; I'll remove the functionality that saves the state in `localStorage`.

Reply:
Done!